source/common/network/connection_impl.cc

```diff
@@ -19,13 +19,12 @@ ConnectionImpl::ConnectionImpl(Event::DispatcherImpl& dispatcher, int fd,
     : filter_manager_(*this, *this), remote_address_(remote_address), dispatcher_(dispatcher),
       fd_(fd), id_(++next_global_id_) {
 
+  // Treat the lack of a valid fd (which in practice only happens if we run out of FDs) as an OOM
+  // condition and just crash.
+  RELEASE_ASSERT(fd_ != -1);
```

Comment:
would it make sense to add a stat check for this first?  or are you 100% sure it never happens?